graphsdk/src/main/java/com/microsoft/graph/model/DateOnly.java

```diff
@@ -0,0 +1,89 @@
+package com.microsoft.graph.model;
+
+
+import java.text.ParseException;
+import java.util.Locale;
+
+/**
+ * A timezone-nonspecific date
+ */
+public class DateOnly {
+
+    private final int
+            mYear,
+            mMonth,
+            mDay;
+
+    /**
+     * Constructs a timezone-nonspecific DateOnly
+     *
+     * @param dateStr date string of the form <code>yyyy-mm-dd</code>
+     */
+    public DateOnly(String dateStr) throws ParseException {
```

Comment:
sorry I missed this before, Make this a static function, `public DateOnly Parse(String ...) throws ParseException {..} `, that returns a DateOnly instance.  We shouldn't be parsing and throwing exceptions in a ctor